src/NServiceBus.AcceptanceTests/Recoverability/Retries/When_fails_with_retries_set_to_0.cs

```diff
@@ -29,37 +30,54 @@ public class Context : ScenarioContext
             public bool GaveUp { get; set; }
         }
 
-        public class RetryEndpoint : EndpointConfigurationBuilder
+        class ErrorNotificationSpy : Feature
         {
-            public RetryEndpoint()
+            protected override void Setup(FeatureConfigurationContext context)
             {
-                EndpointSetup<DefaultServer>(b => b.DisableFeature<Features.SecondLevelRetries>())
-                    .WithConfig<TransportConfig>(c =>
-                    {
-                        c.MaxRetries = 0;
-                    });
+                context.Container.ConfigureComponent<ErrorNotificationSpyTask>(DependencyLifecycle.SingleInstance);
+                context.RegisterStartupTask(b => b.Build<ErrorNotificationSpyTask>());
             }
 
-            class ErrorNotificationSpy : IWantToRunWhenBusStartsAndStops
+            class ErrorNotificationSpyTask : FeatureStartupTask
             {
-                public Context Context { get; set; }
+                readonly BusNotifications busNotifications;
+                readonly Context context;
 
-                public BusNotifications BusNotifications { get; set; }
+                public ErrorNotificationSpyTask(BusNotifications busNotifications, Context context)
+                {
+                    this.busNotifications = busNotifications;
+                    this.context = context;
+                }
 
-                public Task Start(IMessageSession session)
+                protected override Task OnStart(IMessageSession session)
                 {
-                    BusNotifications.Errors.MessageSentToErrorQueue += (sender, message) => Context.GaveUp = true;
+                    busNotifications.Errors.MessageSentToErrorQueue += (sender, message) => context.GaveUp = true;
                     return session.SendLocal(new MessageToBeRetried
                     {
-                        ContextId = Context.Id
+                        ContextId = context.Id
                     });
                 }
 
-                public Task Stop(IMessageSession session)
+                protected override Task OnStop(IMessageSession session)
                 {
                     return Task.FromResult(0);
                 }
             }
+        }
+
+        public class RetryEndpoint : EndpointConfigurationBuilder
+        {
+            public RetryEndpoint()
+            {
+                EndpointSetup<DefaultServer>(b => {
+                    b.DisableFeature<Features.SecondLevelRetries>();
+                    b.EnableFeature<ErrorNotificationSpy>();
+                } )
+                    .WithConfig<TransportConfig>(c =>
```

Comment:
since the tests seems to test this config option, it would probably make sense to explicitly enable FLR.